openmdao/docs/openmdao_book/features/core_features/running_your_models/set_input_defaults.ipynb

```diff
@@ -0,0 +1,475 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "p5 = om.Problem()\n",
+    "model = p5.model\n",
+    "\n",
+    "model.add_subsystem('square', SquarePlate(), promotes_inputs=['material'])\n",
+    "model.add_subsystem('circle', CirclePlate(), promotes_inputs=['material'])\n",
+    "\n",
+    "model.set_input_defaults('material', 'steel', units='g', src_shape=(1,))\n",
+    "\n",
+    "try:\n",
+    "    p5.setup()       \n",
+    "except Exception as err:\n",
+    "    print(str(err))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "## How the `set_input_defaults` function differs from the `set_val` function\n",
+    "\n",
+    "While both `set_input_defaults` and `set_val` deal with variable management in OpenMDAO, they have distinct purposes and are used in different contexts.\n",
+    "\n",
+    "- `set_input_defaults` is used at the group level to define default metadata (units and initial value) for promoted inputs, specifically to resolve ambiguity when multiple inputs are promoted to the same name. This is crucial for inputs connected to the automatically generated `_auto_ivc` component.\n",
+    "\n",
+    "  - Used to resolve inconsistencies between Auto-IVC values.\n",
+    "\n",
+    "  - Specifically used at the group level to specify metadata to be assumed when multiple inputs are promoted to the same name. This is required when the promoted inputs have differing units or values.\n",
+    "\n",
+    "\n",
+    "- `set_val` is used at the problem level to set the actual value of a variable, including inputs, outputs, and independent variables. It can handle unit conversions and set values for specific indices in array variables.\n",
+    "\n",
+    "  - Used at the run script level to set the value of an input variable.\n",
+    "\n",
+    "  - Can be used to set the value of a variable in a different unit than its declared unit, and OpenMDAO will perform the conversion.\n",
+    "\n",
+    "  - Can be used to set specific indices or index ranges of array variables.\n",
+    "\n",
+    "In essence, `set_input_defaults` helps OpenMDAO correctly determine the units and initial values of connected inputs during the setup phase, while `set_val` is used to directly manipulate variable values before or during a run.\n",
+    "\n",
+    "*Key Differences*\n",
+    "\n",
+    "-  *Scope*: \n",
+    "   `set_input_defaults` is used at the group level to define default metadata for promoted inputs, while `set_val` is used at the problem level to set specific values for variables.\n",
+    "\n",
+    "- *Purpose*: \n",
+    "  `set_input_defaults` resolves ambiguities when multiple inputs are promoted to the same name, while `set_val` is used to assign values to variables.\n",
+    "\n",
+    "- *Timing*: \n",
+    "  `set_input_defaults` is typically called during the model setup phase, while `set_val` can be called before or during a run of the model."
+   ]
```

Comment:
"- `set_val` is used at the problem level to set the actual value of a variable, including inputs, outputs, and independent variables."

We might want to also mention that initial values for implicit states should be set with `set_val`. Not sure what independent variables are here, and how they differ from inputs, so maybe we just say "inputs and implicit states".